### PR TITLE
Migration : Frontend

### PR DESCRIPTION
**Pull request by Nazi-pikachu**
## Description
To migrate the frontend from previous seperate repo to the main repo

## Changes Made
Generated a react-app using nx

## Related Issues
- [x] #76 
- [ ]  #52 

## Testing Instructions

1. Follow the setup readme 
2. run `nx serve frontend`

## Screenshots (if applicable)
Current output 
image

## Additional Notes
Its an intial PR in which the frontend code has been migrated to frontend app follow test instructions to know how to run the frontend.

### CODE REVIEW

File: apps/frontend/src/app/Protect.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
+
+import { useNavigate } from "react-router-dom";
+
+function Protect(props) {
+  let Cmp = props.Cmp;
```
Comment (yuvrajsab) [inline apps/frontend/src/app/Protect.js:6]: use proper names for variables.

File: apps/frontend/src/app/apis/index.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+export const mockUser = {
+  name: "Test User",
+  email: "test@gmail.com",
+  password: "test@123",
+};
+const baseUrl = "https://fa.chakshu-rd.samagra.io/api";
```
Comment (yuvrajsab) [inline apps/frontend/src/app/apis/index.js:6]: access these from an env or app constants.

File: apps/frontend/src/app/apis/index.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      registration: {
+        applicationId: "650c50b5-eae9-442e-88de-fc36be10a21b", // TO DO : Change this to env variable
+      },
+      user: {
+        name: "test",
+        email: "test@gmail.com",
+        password: "test@123",
+      },
+    })
+  );
```
Comment (yuvrajsab) [inline apps/frontend/src/app/apis/index.js:19]: remove unnecessary logs

File: apps/frontend/src/app/assets/styles/responsive.css [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    left: auto;
+  }
+  .layout-dashboard-rtl .ant-layout {
+    margin-right: 270px;
+    margin-left: 0;
+  }
+  .layout-dashboard-rtl .ant-layout-footer {
+    margin: 0 20px 20px 0;
+  }
+}
```
Comment (yuvrajsab) [inline apps/frontend/src/app/assets/styles/responsive.css:67]: min widths should come in ascending order. e.x 768 then 992 then 1024.

File: apps/frontend/src/app/assets/styles/responsive.css [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    margin: 0 7px;
+  }
+  .layout-dashboard .header-control .btn-sign-in span {
+    display: inline;
+  }
+  .profile-nav-bg {
+    margin-top: -87.8px;
+  }
+  .card-profile-head {
+    margin: -53px 24px 24px;
```
Comment (yuvrajsab) [inline apps/frontend/src/app/assets/styles/responsive.css:37]: negative margins should be avoided.

File: apps/frontend/src/app/pages/FormHomePage.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+function FormHomePage() {
+  const [state, setState] = useState({
+    userID: "0fe6ff38-fc46-11ec-b939-0242ac120001",
+    url: "",
+    project: "0fe6ff38-fc46-11ec-b939-0242ac120002",
+    customHashId: nanoid(6),
+    titleImage: "",
+    urlImg: "",
+    description: "",
```
Comment (yuvrajsab) [inline apps/frontend/src/app/pages/FormHomePage.js:14]: should not be hardcoded like that.

File: apps/frontend/src/app/pages/FormTagsPage.js [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+              <Footer>
+                <br></br>
+                <div class="flexbox-container" justify-content="space-between">
+                  &nbsp;&nbsp;
+                  <Interaction
+                    icon={LOVE_ICON}
+                    count={12}
+                    style={{ paddingTop: "10px" }}
+                  />
+                  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (yuvrajsab) [inline apps/frontend/src/app/pages/FormTagsPage.js:238]: find another way to add spaces. remove this &nbsp;

**Comment by Nazi-pikachu** [comment]
Progress on this PR Till now

- Frontend Migrated to monorepo
- Migrated to react-router-dom v6
- Fixed few pages for responsiveness
- Refactored the entire Frontend as per the standards
- Deployed to vercel to ease tracking

File: apps/frontend/src/app/apis/index.js (hunk 2 above)
Comment (Shruti3004) [reply apps/frontend/src/app/apis/index.js:1]: @Nazi-pikachu

File: apps/frontend/src/app/Protect.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
```
Comment (Shruti3004) [inline apps/frontend/src/app/Protect.js:1]: @Nazi-pikachu can we move this in folder name `utils`

File: apps/frontend/src/app/app.jsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { Routes, Route, Navigate } from "react-router-dom";
+import SignIn from "app/pages/Auth/SignIn";
+import SignUp from "app/pages/Auth/SignUp";
+import Main from "app/components/Main";
+function App() {
+  return (
+    <div className="App">
+      <Routes>
+        <Route path="/sign-up" element={<SignUp />} />
```
Comment (Shruti3004) [inline apps/frontend/src/app/app.jsx:9]: @Nazi-pikachu please create the constants folder for all the routes

File: apps/frontend/src/app/app.module.css [hunk 10]
```diff
@@ -0,0 +1 @@
+/* Your styles goes here. */
```
Comment (Shruti3004) [inline apps/frontend/src/app/app.module.css:1]: Delete this file

File: apps/frontend/src/app/assets/constants/Config/DashboardConfig/index.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import { piedata } from "app/assets/constants/mockData";
```
Comment (Shruti3004) [inline apps/frontend/src/app/assets/constants/Config/DashboardConfig/index.js:1]: @Nazi-pikachu please take the constants and styles folder out of it

**Review by Shruti3004 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/Protect.js (hunk 8 above)
Comment (Nazi-pikachu) [reply apps/frontend/src/app/Protect.js:1]: Resolved

File: apps/frontend/src/app/app.module.css (hunk 10 above)
Comment (Nazi-pikachu) [reply apps/frontend/src/app/app.module.css:1]: resolved

File: apps/frontend/src/app/assets/constants/Config/DashboardConfig/index.js (hunk 11 above)
Comment (Nazi-pikachu) [reply apps/frontend/src/app/assets/constants/Config/DashboardConfig/index.js:1]: resolved

File: apps/frontend/src/app/app.jsx (hunk 9 above)
Comment (Nazi-pikachu) [reply apps/frontend/src/app/app.jsx:9]: resolved

**Pull request merged**